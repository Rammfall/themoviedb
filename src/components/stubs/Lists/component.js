import React from 'react'
import {
  Layout, Row, Col, Card, Typography, Modal, Pagination
} from 'antd'
import { PlusCircleOutlined, DeleteOutlined } from '@ant-design/icons'
import { range } from 'lodash'

import Header from '../Header'
import CreateListModal from '../CreateListModal'

const showModal = () => {
  Modal.confirm({
    title: 'Do you want to delete list?',
    onOk() {},
    onCancel() {}
  })
}

class Lists extends React.Component {
  constructor(props) {
    super(props)

    this.state = { modalVisible: false }

    this.showModal = () => {
      this.setState({ modalVisible: true })
    }

    this.hideModal = () => {
      this.setState({ modalVisible: false })
    }
  }

  render() {
    const { modalVisible } = this.state

    return (
      <Layout>
        <Header />
        <Layout.Content>
          <Row>
            <Col
              offset={2}
              span={20}
            >
              <div className="top-margin">
                <Typography.Title>
                  My Lists
                  {' '}
                  <PlusCircleOutlined onClick={this.showModal} />
                </Typography.Title>
              </div>
            </Col>
          </Row>
          <Row
            gutter={8}
            type="flex"
          >
            <Col
              span={20}
              offset={2}
            >
              <Row
                gutter={{
                  xs: 8, sm: 16, md: 24, lg: 32
                }}
              >
                {range(10).map(index => (
                  <Col
                    key={index}
                    xs={{ span: 24 }}
                    sm={{ span: 12 }}
                    md={{ span: 8 }}
                    lg={{ span: 8 }}
                    xl={{ span: 6 }}
                  >
                    <Card
                      hoverable
                      className="top-margin"
                      actions={[
                        <DeleteOutlined
                          key="delete"
                          onClick={showModal}
                        />
                      ]}
                    >
                      <Typography.Title level={4}>
                      List name
                        {index}
                      </Typography.Title>
                    Description
                    </Card>
                  </Col>
                ))}
              </Row>
            </Col>
          </Row>
          <Row
            type="flex"
            justify="center"
          >
            <Col>
              <Pagination
                defaultCurrent={1}
                total={50}
                className="pagination"
              />
            </Col>
          </Row>
        </Layout.Content>
        <Modal
          visible={modalVisible}
          onCancel={this.hideModal}
          okText="Create"
          title="Create list"
        >
          <CreateListModal />
        </Modal>
      </Layout>
    )
  }
}

export default Lists
