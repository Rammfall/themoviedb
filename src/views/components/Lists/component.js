import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row, Typography } from 'antd'
import { PlusCircleOutlined } from '@ant-design/icons'
import { useIntl } from 'react-intl'

import PrivateLayout from 'Views/layouts/PrivateLayout'
import RenderCtrl from 'Views/components/RenderCtrl'
import Loader from 'Views/components/Loader'
import EmptyState from 'Views/components/EmptyState'
import ListsLists from './ListsLists'
import CreateListModal from './CreateListModal'

const Lists = ({
  isEmpty,
  isLoading,
  createModalVisibility,
  toggleCreateModal
}) => {
  const { formatMessage } = useIntl()

  return (
    <PrivateLayout>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className="top-margin">
            <Typography.Title>
              {formatMessage({ id: 'lists.title' })}
              <PlusCircleOutlined onClick={toggleCreateModal} />
            </Typography.Title>
          </div>
        </Col>
      </Row>
      <RenderCtrl
        isEmpty={isEmpty}
        isLoading={isLoading}
        renderList={<ListsLists />}
        renderLoading={<Loader />}
        renderEmpty={<EmptyState title={formatMessage({ id: 'lists.empty' })} />}
      />
       <CreateListModal
         isVisible={createModalVisibility}
         toggleHandler={toggleCreateModal}
       />
    </PrivateLayout>
  )
}

Lists.propTypes = {
  isEmpty: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool,
  createModalVisibility: PropTypes.bool.isRequired,
  toggleCreateModal: PropTypes.func.isRequired
}

Lists.defaultProps = {
  isLoading: false
}

export default Lists
