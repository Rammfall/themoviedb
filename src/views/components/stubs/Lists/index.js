import Lists from './component'

export default Lists
