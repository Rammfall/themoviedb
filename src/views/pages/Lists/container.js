import React, { Component } from 'react'
import { compose } from 'ramda'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import { getListsSelector, isEmptyListsSelector } from 'Store/concepts/lists/selectors'
import { loadingSelector } from 'Store/concepts/data/selectors'
import { lists as listsConstant } from 'Store/concepts/lists/endpoints'
import { getLists } from 'Store/concepts/lists/actions'

import Lists from 'Views/components/Lists'
import { userIdSelector } from 'Store/concepts/account/selectors'
import queryParams from 'Utils/router/queryParams'

class ListsPage extends Component {
  componentDidMount() {
    this.getData()
  }

  componentDidUpdate(prevProps) {
    const { userId, location: { search } } = this.props

    if(userId !== prevProps.userId || search !== prevProps.location.search) {
      this.getData()
    }
  }

  getData = () => {
    const { getAllLists, userId, location: { search } } = this.props
    const page = queryParams(search).get('page')

    if (userId) {
      getAllLists(page)
    }
  }

  render() {
    const { isEmpty, isLoading, lists } = this.props

    return (
      <Lists
        isEmpty={isEmpty}
        isLoading={isLoading}
        lists={lists}
      />
    )
  }
}

ListsPage.propTypes = {
  isEmpty: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool,
  lists: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string
  })).isRequired,
  userId: PropTypes.number,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired
  }).isRequired,
  getAllLists: PropTypes.func.isRequired
}

ListsPage.defaultProps = {
  isLoading: false,
  userId: null
}

const mapStateToProps = (state) => ({
  lists: getListsSelector(state),
  isEmpty: isEmptyListsSelector(state),
  isLoading: loadingSelector(state, listsConstant),
  userId: userIdSelector(state)
})

const mapDispatchToProps = {
  getAllLists: getLists
}

export { ListsPage as ListsPageContainer }
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter
)(ListsPage)
