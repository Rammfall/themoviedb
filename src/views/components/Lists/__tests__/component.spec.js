import React from 'react'
import { shallow } from 'enzyme'

import Lists from '../component'

jest.mock('react-intl', () => ({
  useIntl: jest.fn(() => ({
    formatMessage: jest.fn(() => 'test')
  }))
}))

describe('Lists', () => {
  it('matches snapshot', () => {
    const props = {
      isEmpty: false,
      isLoading: false,
      lists: [{
        name: 'test',
        description: 'test',
        id: 3
      }],
      createModalVisibility: false,
      toggleCreateModal: jest.fn
    }
    const component = shallow(<Lists {...props} />)

    expect(component).toMatchSnapshot()
  })
})
