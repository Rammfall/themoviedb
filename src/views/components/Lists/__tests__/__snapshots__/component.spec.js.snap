// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists matches snapshot 1`] = `
<PrivateLayout>
  <Row>
    <Col
      offset={2}
      span={20}
    >
      <div
        className="top-margin"
      >
        <Title>
          test
          <ForwardRef(PlusCircleOutlined)
            onClick={[Function]}
          />
        </Title>
      </div>
    </Col>
  </Row>
  <RenderCtrl
    isEmpty={false}
    isLoading={false}
    renderEmpty={
      <EmptyState
        title="test"
      />
    }
    renderList={<Unknown />}
    renderLoading={<Loader />}
  />
  <Connect(withRouter(WithFormik(CreateListModal)))
    isVisible={false}
    toggleHandler={[Function]}
  />
</PrivateLayout>
`;
